### PR TITLE
[DYN-41]feat: add sell/buy crypto and congratulations

### PR DESCRIPTION
**Pull request by liubaterehova**

### CODE REVIEW

File: src/components/DailyTransactions/DailyTransactions.jsx [hunk 1]
```diff
@@ -26,18 +27,24 @@ function DailyTransactions({ date, transactions }) {
...
+        const total = await getDailyTotal(start, end);
+        setTotalAmount(total);
+      };
 
-    init();
+      init();
+    } finally {
+      setLoading(false);
+    }
   }, [date, transactions.length]);
```
Comment (arlovaalina) [inline src/components/DailyTransactions/DailyTransactions.jsx:43]: Do we really need `transactions.length` in dependencies?

File: src/helpers/constants.js [hunk 2]
```diff
@@ -128,3 +128,12 @@ export const ANALYTICS_EVENTS = {
   // CHOOSE_MERCHANT_OR_ITEM: 'Payment - Choose Merchant or Item',
   PAYMENT_COMPLETE_TRANSACTION: 'Payment - Complete transaction',
 };
+
+const { height } = Dimensions.get('window');
+
+export const KEYBOARD_VERTICAL_OFFSET = () => {
```
Comment (arlovaalina) [inline src/helpers/constants.js:134]: Does it work? Now it is a function, but you use it as a simple value. Previously it was wrapped in `useMemo`, that's why we could use it as a value.

File: src/screens/CryptoOperations/components/ConvertCard/ConvertCard.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  realMoney,
+  cryptoMoney,
+  fromRealToCrypto,
+  handleChangeRealMoney,
+  handleChangeCrypto,
+}) {
+  const RealMoneyIcon = getCurrencyIcon(CURRENCIES.REAL.FRANK);
+  const CryptoIcon = getCurrencyIcon(crypto);
+
+  let convertFrom = (
```
Comment (arlovaalina) [inline src/screens/CryptoOperations/components/ConvertCard/ConvertCard.jsx:26]: Hmm, it is not a good practice to assign jsx to variable in such way. If you do that, you should think about creating a new component.

**Review by arlovaalina — CHANGES_REQUESTED** [review]

File: src/components/ConfirmDeposit/ConfirmDeposit.jsx [hunk 4]
```diff
@@ -70,21 +62,14 @@ function ConfirmDeposit({
...
-
   return (
     <>
       {isSubmitting && <FullScreenLoader />}
       <SafeAreaView style={styles.screen}>
         <KeyboardAvoidingView
           behavior={Platform.OS === 'android' ? 'height' : 'padding'}
           style={styles.container}
-          keyboardVerticalOffset={keyboardVerticalOffset}
+          keyboardVerticalOffset={KEYBOARD_VERTICAL_OFFSET}
```
Comment (arlovaalina) [inline src/components/ConfirmDeposit/ConfirmDeposit.jsx:72]: Here you use constant as a value, but actually it is a function. I think you should choose: either leave it as a function (but then it is better to remove it from constants and give it another name starting with a verb) or make from it a value.

**Review by arlovaalina — CHANGES_REQUESTED** [review]

**Review by apsevkin — CHANGES_REQUESTED** [review]

File: src/screens/CryptoOperations/components/ConvertCards/ConvertCards.jsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      />
+      <Card
+        title="You get"
+        currency={cards.to.currency}
+        setValue={(e) => handleChangeMoney(e, cards.to.currency)}
+        value={cards.to.value}
+      />
+      <TouchableOpacity
+        onPress={() =>
+          setCards((state) => ({ from: state.to, to: state.from }))
```
Comment (arlovaalina) [inline src/screens/CryptoOperations/components/ConvertCards/ConvertCards.jsx:63]: And this

File: src/screens/CryptoOperations/components/ConvertCards/ConvertCards.jsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      <Card
+        title="You spend"
+        currency={cards.from.currency}
+        setValue={(e) => handleChangeMoney(e, cards.from.currency)}
+        value={cards.from.value}
+      />
+      <Card
+        title="You get"
+        currency={cards.to.currency}
+        setValue={(e) => handleChangeMoney(e, cards.to.currency)}
```
Comment (arlovaalina) [inline src/screens/CryptoOperations/components/ConvertCards/ConvertCards.jsx:58]: And this

File: src/screens/CryptoOperations/components/ConvertCards/ConvertCards.jsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  value: PropTypes.string.isRequired,
+};
+
+function ConvertCard({ cards, setCards, handleChangeMoney }) {
+  return (
+    <>
+      <Card
+        title="You spend"
+        currency={cards.from.currency}
+        setValue={(e) => handleChangeMoney(e, cards.from.currency)}
```
Comment (arlovaalina) [inline src/screens/CryptoOperations/components/ConvertCards/ConvertCards.jsx:52]: Let's move it into separate function.

File: src/screens/CryptoOperations/components/ConvertCards/ConvertCards.jsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+Card.propTypes = {
+  title: PropTypes.string.isRequired,
+  currency: PropTypes.string.isRequired,
+  setValue: PropTypes.func.isRequired,
+  value: PropTypes.string.isRequired,
+};
+
+function ConvertCard({ cards, setCards, handleChangeMoney }) {
```
Comment (arlovaalina) [inline src/screens/CryptoOperations/components/ConvertCards/ConvertCards.jsx:46]: Why do you have two components in one file?

File: src/screens/CryptoOperations/CryptoOperations.jsx [hunk 9]
```diff
@@ -28,22 +33,65 @@ const getTitleHeader = (operation, crypto) => {
...
+Big.strict = true;
 
 function CryptoOperations({ route, navigation }) {
   const { operation, crypto } = route.params;
-  const [cryptoMoney, setCryptoMoney] = useState('0');
-  const [realMoney, setRealMoney] = useState('0');
-  const [balanceRealMoney, setBalanceRealMoney] = useState(0);
-  const [balanceCryptoMoney, setBalanceCryptoMoney] = useState(10);
-  const [fromRealToCrypto, setFromRealToCrypto] = useState(true);
+  const cardsValues = {
```
Comment (arlovaalina) [inline src/screens/CryptoOperations/CryptoOperations.jsx:40]: Let's wrap it into `useMemo`, otherwise each render will create a new object.

**Review by arlovaalina — CHANGES_REQUESTED** [review]

**Review by arlovaalina — APPROVED** [review]

**Pull request merged**